apps/emqx_eviction_agent/test/emqx_eviction_agent_SUITE.erl

```diff
@@ -0,0 +1,232 @@
+       emqx_eviction_agent:status()),
+
+    ok = emqx_eviction_agent:disable(test_eviction).
+
+
+t_explicit_session_takeover(Config) ->
+    erlang:process_flag(trap_exit, true),
+
+    {ok, C0} = emqtt_connect(<<"client_with_session">>, false),
+    {ok, _, _} = emqtt:subscribe(C0, <<"t1">>),
+
+    ok = emqx_eviction_agent:enable(test_eviction, undefined),
+
+    ?assertEqual(
+       1,
+       emqx_eviction_agent:connection_count()),
+
+    ok = emqx_eviction_agent:evict_connections(1),
+
+    receive
+        {'EXIT', C0, {disconnected, ?RC_USE_ANOTHER_SERVER, _}} -> ok
+    after 1000 ->
+              ?assert(false, "Connection not evicted")
+    end,
+
+    ?assertEqual(
+       0,
+       emqx_eviction_agent:connection_count()),
+
+    ?assertEqual(
+       1,
+       emqx_eviction_agent:session_count()),
+
+    %% First, evacuate to the same node
+
+    ?check_trace(
+       ?wait_async_action(
+          emqx_eviction_agent:evict_sessions(1, node()),
+          #{?snk_kind := emqx_channel_takeover_end},
+          1000),
+       fun(_Result, Trace) ->
+        ?assertMatch(
+           [#{clientid := <<"client_with_session">>} | _ ],
+           ?of_kind(emqx_channel_takeover_end, Trace))
+       end),
+
+    ok = emqx_eviction_agent:disable(test_eviction),
+    ok = connect_and_publish(<<"t1">>, <<"MessageToEvictedSession1">>),
+    ok = emqx_eviction_agent:enable(test_eviction, undefined),
+
+    %% Evacuate to another node
+
+    TargetNodeForEvacuation = ?config(evacuate_node, Config),
+    ?check_trace(
+       ?wait_async_action(
+          emqx_eviction_agent:evict_sessions(1, TargetNodeForEvacuation),
+          #{?snk_kind := emqx_channel_takeover_end},
+          1000),
+       fun(_Result, Trace) ->
+        ?assertMatch(
+           [#{clientid := <<"client_with_session">>} | _ ],
+           ?of_kind(emqx_channel_takeover_end, Trace))
+       end),
+
+    ?assertEqual(
+       0,
+       emqx_eviction_agent:session_count()),
+
+    ?assertEqual(
+       1,
+       rpc:call(TargetNodeForEvacuation, emqx_eviction_agent, session_count, [])),
+
+    ok = emqx_eviction_agent:disable(test_eviction),
+
+    ct:pal("evicted chann info: ~p", [emqx_cm:get_chan_info(<<"client_with_session">>)]),
+
+    ok = connect_and_publish(<<"t1">>, <<"MessageToEvictedSession2">>),
+    ct:sleep(100),
+
+    {ok, C2} = emqtt_connect(<<"client_with_session">>, false),
+
+    ok = assert_receive_publish(
+           [#{payload => <<"MessageToEvictedSession1">>, topic => <<"t1">>},
+            #{payload => <<"MessageToEvictedSession2">>, topic => <<"t1">>}]),
+    ok = emqtt:disconnect(C2).
+
+t_disable_on_restart(_Config) ->
+    ok = emqx_eviction_agent:enable(test_eviction, undefined),
+
+    ok = supervisor:terminate_child(emqx_eviction_agent_sup, emqx_eviction_agent),
+    {ok, _} = supervisor:restart_child(emqx_eviction_agent_sup, emqx_eviction_agent),
+
+    ?assertEqual(
+        disabled,
+        emqx_eviction_agent:status()).
+
```

Comment:
Q: would it make sense to add a test that asserts that we cannot evict connections/sessions to another node that is also being evacuated?

Reply:
✅